File: src/_1ProgrammingBasicsWithJavaMay2023/_3ConditionalStatementsAdvanced/_2Exercise/_03NewHouse.java
package _1ProgrammingBasicsWithJavaMay2023._3ConditionalStatementsAdvanced._2Exercise;

import java.util.Scanner;

public class _03NewHouse {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String flowers = scanner.nextLine();
        int numberOfFlowers = Integer.parseInt(scanner.nextLine());
        int budget = Integer.parseInt(scanner.nextLine());
        double priceOfRoses = 5.00;
        double priceOfDahlias = 3.80;
        double priceOfTulips = 2.80;
        double priceOfNarcissus = 3.00;
        double priceOfGladiolus = 2.50;
        double finalPrice = 0;
        switch (flowers) {
            case "Roses":
                finalPrice = priceOfRoses * numberOfFlowers;
                if (numberOfFlowers > 80) {
                    finalPrice = finalPrice * 0.90;
                }
                break;
            case "Dahlias":
                finalPrice = priceOfDahlias * numberOfFlowers;
                if (numberOfFlowers > 90) {
                    finalPrice = finalPrice * 0.85;
                }
                break;
            case "Tulips":
                finalPrice = priceOfTulips * numberOfFlowers;
                if (numberOfFlowers > 80) {
                    finalPrice = finalPrice * 0.85;
                }
                break;
            case "Narcissus":
                finalPrice = priceOfNarcissus * numberOfFlowers;
                if (numberOfFlowers < 120) {
                    finalPrice = finalPrice * 1.15;
                }
                break;
            case "Gladiolus":
                finalPrice = priceOfGladiolus * numberOfFlowers;
                if (numberOfFlowers < 80) {
                    finalPrice = finalPrice * 1.20;
                }
                break;
        }
        double diff =Math.abs(budget - finalPrice);
        if (budget >= finalPrice) {
            System.out.printf("Hey, you have a great garden with %d %s and %.2f leva left.", numberOfFlowers, flowers, diff );
        } else {
            System.out.printf("Not enough money, you need %.2f leva more.", diff);
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_3ConditionalStatementsAdvanced/_2Exercise/_03NewHouse.js
function newHouse(flowers, numberOfFlowers, budget) {
    const ROSES_PRICE = 5.00;
    const DAHLIAS_PRICE = 3.80;
    const TULIPS_PRICE = 2.80;
    const NARCISSUS_PRICE = 3.00;
    const GLADIOLUS_PRICE = 2.50;

    const DISCOUNT_IF_BUY_MORE_THAN_80_ROSES = 0.10;
    const DISCOUNT_IF_BUY_MORE_THAN_90_DAHLIAS = 0.15;
    const DISCOUNT_IF_BUY_MORE_THAN_80_TULIPS = 0.15;
    const INCREASE_IF_BUY_LESS_THAN_120_NARCISSUS = 0.15;
    const INCREASE_IF_BUY_LESS_THAN_80_GLADIOLUS = 0.20;

    let totalSum = 0.0;

    if (flowers === "Roses") {
        totalSum = numberOfFlowers * ROSES_PRICE;
        if (numberOfFlowers > 80) {
            totalSum = totalSum - (totalSum * DISCOUNT_IF_BUY_MORE_THAN_80_ROSES);
        }
    } else if (flowers === "Dahlias") {
        totalSum = numberOfFlowers * DAHLIAS_PRICE;
        if (numberOfFlowers > 90) {
            totalSum = totalSum - (totalSum * DISCOUNT_IF_BUY_MORE_THAN_90_DAHLIAS);
        }
    } else if (flowers === "Tulips") {
        totalSum = numberOfFlowers * TULIPS_PRICE;
        if (numberOfFlowers > 80) {
            totalSum = totalSum - (totalSum * DISCOUNT_IF_BUY_MORE_THAN_80_TULIPS);
        }
    } else if (flowers === "Narcissus") {
        totalSum = numberOfFlowers * NARCISSUS_PRICE;
        if (numberOfFlowers < 120) {
            totalSum = totalSum + (totalSum * INCREASE_IF_BUY_LESS_THAN_120_NARCISSUS);
        }
    } else if (flowers === "Gladiolus") {
        totalSum = numberOfFlowers * GLADIOLUS_PRICE;
        if (numberOfFlowers < 80) {
            totalSum = totalSum + (totalSum * INCREASE_IF_BUY_LESS_THAN_80_GLADIOLUS);
        }
    }

    const diff = Math.abs(budget - totalSum);

    if (budget >= totalSum) {
        console.log(`Hey, you have a great garden with ${numberOfFlowers} ${flowers} and ${diff.toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money, you need ${diff.toFixed(2)} leva more.`);
    }

}